src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,236 @@
+
+The conventional way to linearise the Jordan axiom is to equate coefficients (more formally, assume
+that the axiom holds in all field extensions). For simplicity we use brute force algebraic expansion
+and substitution instead.
+
+## Motivation
+
+Every Jordan algebra `A` has a triple product defined, for `a` `b` and `c` in `A` by
+$$
+{a\,b\,c} = (a * b) * c - (a * c) * b + a * (b * c).
+$$
+Via this triple product Jordan algebras are related to a number of other mathematical structures:
+Jordan triples, partial Jordan triples, Jordan pairs and quadratic Jordan algebras. In addition to
+their considerable algebraic interest ([mccrimmon2004]) these structures have been shown to have
+deep connections to mathematical physics, functional analysis and differential geometry. For more
+information about these connections the interested reader is referred to [alfsenshultz2003],
+[chu2012], [friedmanscarr2005], [iordanescu2003] and [upmeier1987].
+
+Every associative algebra can be equipped with a symmetrized multiplication (characterized by
+`sym_alg.sym_mul_sym`) making it into a commutative Jordan algebra. Jordan algebras arising this way
+are said to be special.
+
+There are also exceptional Jordan algebras which can be shown not to be the symmetrization of any
+associative algebra. The 3x3 matrices of octonions is the canonical example.
+
+Non-commutative Jordan algebras have connections to the Vidav-Palmer theorem
+[cabreragarciarodriguezpalacios2014].
+
+## References
+
+* [Cabrera García and Rodríguez Palacios, Non-associative normed algebras. Volume 1]
+  [cabreragarciarodriguezpalacios2014]
+* [Hanche-Olsen and Størmer, Jordan Operator Algebras][hancheolsenstormer1984]
+* [McCrimmon, A taste of Jordan algebras][mccrimmon2004]
+
+-/
+
+variables (A : Type*)
+
+/-- A (non-commutative) Jordan multiplication. -/
+class is_jordan [has_mul A] :=
+(lmul_comm_rmul : ∀ a b : A, (a * b) * a = a * (b * a))
+(lmul_lmul_comm_lmul : ∀ a b : A, (a * a) * (a * b) = a * ((a * a) * b))
+(lmul_lmul_comm_rmul : ∀ a b : A, (a * a) * (b * a) = ((a * a) * b) * a)
+(lmul_comm_rmul_rmul : ∀ a b : A, (a * b) * (a * a) = a * (b * (a * a)))
+(rmul_comm_rmul_rmul : ∀ a b : A, (b * a) * (a * a) = (b * (a * a)) * a)
+
+/-- A commutative Jordan multipication -/
+class is_comm_jordan [has_mul A] :=
+(mul_comm : ∀ a b : A, a * b = b * a)
+(jordan : ∀ a b : A, (a * b) * (a * a) = a * (b * (a *a)))
+
+/-- A (commutative) Jordan multiplication is also a Jordan multipication -/
+@[priority 100] -- see Note [lower instance priority]
+instance is_comm_jordan.to_is_jordan [has_mul A] [is_comm_jordan A] : is_jordan A :=
+{ lmul_comm_rmul := λ a b, by rw [is_comm_jordan.mul_comm, is_comm_jordan.mul_comm a b],
+  lmul_lmul_comm_lmul := λ a b, by rw [is_comm_jordan.mul_comm (a * a) (a * b),
+    is_comm_jordan.jordan, is_comm_jordan.mul_comm b (a * a)],
+  lmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm, ←is_comm_jordan.jordan,
```

Comment:
Can't you prove this directly with `is_comm_jordan.jordan`? The fields look identical to me.

Reply:
It seems so.